libmikmod/playercode/mplayer.c

```diff
@@ -1613,21 +1719,18 @@ static int DoXMEffectL(UWORD tick, UWORD flags, MP_CONTROL *a, MODULE *mod, SWOR
 	dat=UniGetByte();
 	if ((!tick)&&(a->main.i)) {
 		INSTRUMENT *i=a->main.i;
-		MP_VOICE *aout;
+		MP_VOICE *aout = &mod->voice[channel];
```

Comment:
Not sure what exactly is going on with MikMod's handling of channel assignment; is there a reason this was changed from `a->slave`?

Reply:
It is a while ago, since I made this patch, so I cannot exactly remember. I think it is because a->slave is null, since it is a XM effect, so I need to get the voice in a different way.